data-contracts/README.md

```diff
@@ -0,0 +1,439 @@
+# Managing Data Contracts
+
+Data contracts consist not only of the schemas to define the structure of events, but also rulesets allowing for more fine-grained validations,
+controls, and discovery. In this tutorial, we'll evolve a schemas and add data quality and migration rules.
+
+<img src="./images/overview.png" width="1000" height="600">
+
+In the workflow above, we see these tools in action:
+* The [Confluent Terraform Provider](https://registry.terraform.io/providers/confluentinc/confluent/latest/docs) is used to define Confluent Cloud assets (Kafka cluster(s), Data Governance, Kafka Topics, and Schema Configurations).
+* Using the newly created Schema Registry, data engineers and architects define the schema of the events that comprise the organization's canonical data model - i.e. entities, events, and commands that are shared across applications. - along with other parts of the data contract. This includes data quality rules, metadata, and migration rules. A gradle plugin is utilized to register the schemas and related elements of the data contract with the Schema Registry.
+* Applications which producer and/or consume these event types can download the schemas from the Schema Registry. In our example, this is a JVM application built using Gradle. A gradle plugin is used to download the schemas, after which another gradle plugin is used to generate Java classes from those schemas - thus providing the application with compile-time type safety.
+
+We will cover these steps in detail.
+
+## Running the Example
+
+In this tutorial we'll create Confluent Cloud infrastructure - including a Kafka cluster and Schema Registry. Then we'll create
+a Kafka topic named `membership-avro` to store `Membership` events. The Apache Avro schema is maintained and managed in this repo
+along with metadata and migration rules. 
+
+We will evolve the `membership` schema, refactoring the events to encapsulate the date-related fields of version 1 into its
+own `record` type in version 2. Typically this would be a breaking change. However, [data migration rules](https://docs.confluent.io/cloud/current/sr/fundamentals/data-contracts.html#migration-rules) in the schema registry
```

Comment:
```suggestion
own `record` type in version 2. Typically this would be a breaking change. However, [data migration rules](https://docs.confluent.io/cloud/current/sr/fundamentals/data-contracts.html#migration-rules) in Schema Registry
```